### PR TITLE
Update deprecated v1beta1 poddisruptionbudget api calls

### PR DESCRIPTION
**Pull request by fossedihelm**
**What this PR does / why we need it**:
Move `policy/v1beta1` PodDisruptionBudget api calls to `policy/v1` becuase `v1beta1` is deprecated in v1.21, and will be unavailable in v1.25. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**: 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xpivarc** [comment]
We cannot merge this as we are supporting 3 releases of Kubernetes.

**Comment by fossedihelm** [comment]
> We cannot merge this as we are supporting 3 releases of Kubernetes.

Thanks. Right, I add `do-not-merge/hold` label for this reason :) k8s 1.23 has been released on 2021-12-07, so IMO it can be merged when we will no longer use 1.20 providers.

**Comment by EdDev** [comment]
The work on just bumping the k8s packages is not suppose to break backward support.
Has something got deprecated that the latest client cannot handler anymore against older k8s API?

@orelmisan , part of this PR is sharing the same work you are looking into. Perhaps you could cooperate on this.

**Comment by fossedihelm** [comment]
> The work on just bumping the k8s packages is not suppose to break backward support. Has something got deprecated that the latest client cannot handler anymore against older k8s API?
>
> @orelmisan , part of this PR is sharing the same work you are looking into. Perhaps you could cooperate on this.

The purpose of the PR is to use `poddisruptionbudget/v1` api instead of `poddisruptionbudget/v1beta1`. `poddisruptionbudget/v1` has been introduced in k8s 1.21, older versions cannot handle it. Hope this can clarify.

**Comment by EdDev** [comment]
> The purpose of the PR is to use `poddisruptionbudget/v1` api instead of `poddisruptionbudget/v1beta1`. `poddisruptionbudget/v1` has been introduced in k8s 1.21, older versions cannot handle it. Hope this can clarify.

Yes, this part I understand, thanks.
I was referring to the first part where the k8s client is bumped, without the usage of the new API.

**Comment by fossedihelm** [comment]
> Yes, this part I understand, thanks. I was referring to the first part where the k8s client is bumped, without the usage of the new API.

Ah ok, sorry, I didn't understand what you were referring to. Thanks. Yes, the bump itself does not have backward compatibility problems

File: tests/migration_test.go
Comment (EdDev) [inline tests/migration_test.go:56]: Why do we keep in tests the beta version?

**Review by EdDev — COMMENTED** [review]
Thank you!
I only had one question regarding the e2e tests.

File: tests/migration_test.go
Comment (fossedihelm) [reply tests/migration_test.go:56]: The `policyv1beta1` is used only  for the `Eviction` type in the `Pods(ns).Evict()` function. Since the purpose of the PR was to update the `poddisruptionbudget` api calls, I think it was misleading mix the things. Furthermore the `Pods(ns).Evict()` has been deprecated from 1.22 in favor of `EvictV1` or `EvictV1beta1` (just for info). What do you think? Thanks

Comment (EdDev) [reply tests/migration_test.go:56]: So you are saying we can remove this in a following PR? (and we do not do it here in order not to mix the two things?)

**Review by EdDev — APPROVED** [review]
Thank you!

**Comment by EdDev** [comment]
I missed this: I do not think you need the release-notes, as you have not touched the API from a (kubevirt) user perspective. All this is internal stuff.

**Comment by fossedihelm** [comment]
> I missed this: I do not think you need the release-notes, as you have not touched the API from a (kubevirt) user perspective. All this is internal stuff.

Thank you, fixed

**Comment by fossedihelm** [comment]
Needs [https://github.com/kubevirt/project-infra/pull/1926](https://github.com/kubevirt/project-infra/pull/1926) to be merged because 1.19 is not compatible

**Comment by xpivarc** [comment]
@fossedihelm Do we already have PR which removes 1.19 lanes?

**Comment by fossedihelm** [comment]
> @fossedihelm Do we already have PR which removes 1.19 lanes?

As mentioned [https://github.com/kubevirt/project-infra/pull/1926](https://github.com/kubevirt/project-infra/pull/1926) we are collecting data from `kind-1.23-vgpu` lane in order to swap them

**Comment by fossedihelm** [comment]
@EdDev puhsed just the rebase, no other changes. Thanks

**Review by EdDev — APPROVED** [review]

**Pull request merged**
